src/app/static/src/app/store/root/actions.ts

```diff
@@ -52,5 +57,27 @@ export const actions: ActionTree<RootState, RootState> & RootActions = {
 
     async changeLanguage(context, payload) {
         await changeLanguage<RootState>(context, payload)
+    },
+
+    async fetchADRKey(context) {
+        await api<RootMutation, RootMutation>(context)
+            .withSuccess(RootMutation.UpdateADRKey)
+            .get("/adr/key/");
+    },
+
+    async saveADRKey(context, key) {
+        await api<RootMutation, RootMutation>(context)
+            .withSuccess(RootMutation.UpdateADRKey)
+            .postAndReturn("/adr/key/", qs.stringify({key}));
+    },
+
+    async deleteADRKey(context) {
+        await api<RootMutation, RootMutation>(context)
+            .delete("/adr/key/")
+            .then((response) => {
```

Comment:
Why not withSuccess as for the others?

Reply:
I was thinking the payload back from the API would be "OK", wile the payload we want to commit is `null`. But actually looking at the controller the payload back from the API will be null, so this _should_ work as `withSuccess`, will change that over.